datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -1138,27 +1138,47 @@ fn numeric_string_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<D
     }
 }
 
+fn coerce_list_children(lhs_field: &FieldRef, rhs_field: &FieldRef) -> Option<FieldRef> {
+    Some(Arc::new(
```

Comment:
Could you add a comment here explaining why this function is needed -- specifically that it is setting the DataType / field name correctly?